everestrs/everestrs/src/lib.rs

```diff
@@ -0,0 +1,196 @@
+mod schema;
+
+use argh::FromArgs;
+use serde::de::DeserializeOwned;
+use std::collections::HashMap;
+use std::os::raw::c_char;
+use std::path::PathBuf;
+use std::pin::Pin;
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error("missing argument to command call: '{0}'")]
+    MissingArgument(&'static str),
+    #[error("invalid argument to command call: '{0}'")]
+    InvalidArgument(&'static str),
+}
+
+pub type Result<T> = ::std::result::Result<T, Error>;
+
+#[cxx::bridge]
+mod ffi {
+    struct CommandMeta {
+        implementation_id: String,
+        name: String,
+        // cxx.rs does not expose a *const c_void type, so we use *const c_char everywhere.
+        // Wherever this shows it means "pointer to some random blob of memory that only Rust
+        // understand what it means and is of no concern to C".
+        obj: *const c_char,
+    }
+
+    struct JsonBlob {
```

Comment:
Not sure about this, but wouldn't a type alias enough here?

Reply:
I do not understand your concerns with this. This is a pretty common pattern in Rust and it does not imbue runtime or compile cost (compared to static functions in C++ classes which need to be mentioned in headers for example).